projects/envoyinit/pkg/runner/run.go

```diff
@@ -30,8 +31,17 @@ const (
 func RunEnvoyValidate(ctx context.Context, envoyExecutable, bootstrapConfig string) error {
 	logger := contextutils.LoggerFrom(ctx)
 
-	validateCmd := cmdutils.Command(ctx, envoyExecutable, "--mode", "validate", "--config-yaml", bootstrapConfig, "-l", "critical", "--log-format", "%v")
-	if err := validateCmd.Run(); err != nil {
+	logger.Debugf("starting full envoy validation with size %d", len(bootstrapConfig))
+
+	validateCmd := cmdutils.Command(ctx, envoyExecutable, "--mode", "validate", "--config-path", "/dev/fd/0",
+		"-l", "critical", "--log-format", "%v")
+	validateCmd = validateCmd.WithStdin(bytes.NewBufferString(bootstrapConfig))
+
+	start := time.Now()
+	err := validateCmd.Run()
+	logger.Debugf("full envoy validation of %d size completed in %s", len(bootstrapConfig), time.Since(start))
```

Comment:
I like this! I wonder if we could make this information consumable to create insights on, so users could see if their configuration was growing over time.

Reply:
I think this might not be the right place but I like the idea. Perhaps higher up in translation we take our final output there to emit a metric?